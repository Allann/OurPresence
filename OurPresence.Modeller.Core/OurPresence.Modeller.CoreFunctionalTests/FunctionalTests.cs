using ApprovalTests;
using OurPresence.Modeller.Interfaces;
using NSubstitute;
using System.Linq;
using OurPresence.Modeller.Generator;
using Xunit;

namespace OurPresence.Modeller.CoreFunctionalTests
{
    public class FunctionalTests
    {
        [Fact]
        public void GivenDataAccessProjectGenerator_WhenGenerating_ThenProjectCreated()
        {
            var packageService = Substitute.For<IPackageService>();
            var settings = Substitute.For<ISettings>();
            settings.SupportRegen.Returns(true);
            var packages = new Packages(packageService, settings);
            settings.Packages.Returns(packages);
            var module = ModuleBuilders.CreateProject();

            var c = new EntityFrameworkProject.Generator(settings, module);

            IProject output = c.Create() as IProject;
            Approvals.VerifyAll(output.FileGroups.SelectMany(f=>f.Files), "file", f => $"{f.FullName}\r\n{f.Content}");
        }

        [Fact]
        public void GivenDataAccessConfigGenerator_WhenSingleSupportRegen_ThenSingleClassCreated()
        {
            var settings = Substitute.For<ISettings>();
            settings.SupportRegen = true;
            var module = ModuleBuilders.CreateModule();

            var c = new EntityFrameworkClass.Generator(settings, module, module.Models.First());

            IFileGroup output = c.Create() as IFileGroup;
            Approvals.VerifyAll(output.Files, "file", f => $"{f.FullName}\r\n{f.Content}");
        }

        [Fact]
        public void GivenDataAccessConfigGenerator_WhenSingleNotSupportRegen_ThenSingleClassCreated()
        {
            var settings = Substitute.For<ISettings>();
            var module = ModuleBuilders.CreateModule();

            var c = new EntityFrameworkClass.Generator(settings, module, module.Models.First());

            IFileGroup output = c.Create() as IFileGroup;
            Approvals.VerifyAll(output.Files, "file", f => $"{f.FullName}\r\n{f.Content}");
        }

        [Fact]
        public void GivenBusinessLogicProjectGenerator_WhenGenerating_ThenProjectCreated()
        {
            var packageService = Substitute.For<IPackageService>();
            var settings = Substitute.For<ISettings>();
            settings.SupportRegen.Returns(true);
            var packages = new Packages(packageService, settings);
            settings.Packages.Returns(packages);
            var module = ModuleBuilders.CreateProject();

            var c = new BusinessLogicProject.Generator(settings, module);

            IProject output = c.Create() as IProject;
            Approvals.VerifyAll(output.FileGroups.SelectMany(f => f.Files), "file", f => $"{f.FullName}\r\n{f.Content}");
        }

        [Fact]
        public void GivenBusinessLogicBehaviourGenerator_WhenSingleSupportRegen_ThenSingleClassCreated()
        {
            var settings = Substitute.For<ISettings>();
            settings.SupportRegen = true;
            var module = ModuleBuilders.CreateModule();

            var c = new BusinessLogicBehaviour.Generator(settings, module, module.Models.First());

            IFileGroup output = c.Create() as IFileGroup;
            Approvals.VerifyAll(output.Files, "file", f => $"{f.FullName}\r\n{f.Content}");
        }

        [Fact]
        public void GivenBusinessLogicBehaviourGenerator_WhenSingleNotSupportRegen_ThenSingleClassCreated()
        {
            var settings = Substitute.For<ISettings>();
            var module = ModuleBuilders.CreateModule();

            var c = new BusinessLogicBehaviour.Generator(settings, module, module.Models.First());

            IFileGroup output = c.Create() as IFileGroup;
            Approvals.VerifyAll(output.Files, "file", f => $"{f.FullName}\r\n{f.Content}");
        }

        [Fact]
        public void GivenApplicationProjectGenerator_WhenGenerating_ThenProjectCreated()
        {
            var packageService = Substitute.For<IPackageService>();
            var settings = Substitute.For<ISettings>();
            settings.SupportRegen.Returns(true);
            var packages = new Packages(packageService, settings);
            settings.Packages.Returns(packages);
            var module = ModuleBuilders.CreateProject();

            var c = new ApplicationProject.Generator(settings, module);

            IProject output = c.Create() as IProject;

            Approvals.VerifyAll(output.FileGroups.SelectMany(f => f.Files), $"file", f => $"{f.FullName}\r\n{f.Content}");
        }

        [Fact]
        public void GivenControllerClassGenerator_WhenSingleSupportRegen_ThenSingleClassCreated()
        {
            var settings = Substitute.For<ISettings>();
            settings.SupportRegen = true;
            var module = ModuleBuilders.CreateModule();

            var c = new ControllerClass.Generator(settings, module, module.Models.First());

            IFileGroup output = c.Create() as IFileGroup;
            Approvals.VerifyAll(output.Files, "file", f => $"{f.FullName}\r\n{f.Content}");
        }

        [Fact]
        public void GivenControllerClassGenerator_WhenSingleNotSupportRegen_ThenSingleClassCreated()
        {
            var settings = Substitute.For<ISettings>();
            var module = ModuleBuilders.CreateModule();

            var c = new ControllerClass.Generator(settings, module, module.Models.First());

            IFileGroup output = c.Create() as IFileGroup;
            Approvals.VerifyAll(output.Files, "file", f => $"{f.FullName}\r\n{f.Content}");
        }

        [Fact]
        public void GivenDomainClassGenerator_WhenSingleNotSupportRegen_ThenSingleClassCreated()
        {
            var settings = Substitute.For<ISettings>();
            var module = ModuleBuilders.CreateModule();

            var c = new DomainClass.Generator(settings, module, module.Models.First());

            IFileGroup output = c.Create() as IFileGroup;
            Approvals.VerifyAll(output.Files, "file", f => $"{f.FullName}\r\n{f.Content}" );
        }

        [Fact]
        public void GivenDomainClassGenerator_WhenSingleSupportRegen_ThenTwoClassedCreated()
        {
            var settings = Substitute.For<ISettings>();
            settings.SupportRegen = true;
            var module = ModuleBuilders.CreateModule();

            var c = new DomainClass.Generator(settings, module, module.Models.First());

            IFileGroup output = c.Create() as IFileGroup;
            Approvals.VerifyAll(output.Files, "file", f => $"{f.FullName}\r\n{f.Content}");
        }

        [Fact]
        public void GivenDomainProjectGenerator_WhenGenerating_ThenProjectCreated()
        {
            var packageService = Substitute.For<IPackageService>();
            var settings = Substitute.For<ISettings>();
            settings.SupportRegen.Returns(true);
            var packages = new Packages(packageService, settings);
            settings.Packages.Returns(packages);
            var module = ModuleBuilders.CreateProject();

            var c = new DomainProject.Generator(settings, module);

            IProject output = c.Create() as IProject;
            Approvals.VerifyAll(output.FileGroups.SelectMany(f => f.Files), "file", f => $"{f.FullName}\r\n{f.Content}");
        }
    }
}
