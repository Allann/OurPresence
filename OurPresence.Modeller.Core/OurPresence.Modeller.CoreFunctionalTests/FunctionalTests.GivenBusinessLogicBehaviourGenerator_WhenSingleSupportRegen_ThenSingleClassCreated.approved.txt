file[0] = Account\Withdraw\Account.generated.cs
// Auto-generated using OurPresence.Modeller template 'Business Logic Behaviour' version 1.0.0

// --------------------------------------------------------------------------------
// WARNING: This file will be overwritten if re-generation is triggered.
// --------------------------------------------------------------------------------

namespace Mizrael.SuperSafeBank.BusinessLogic.Account.Withdraw
{
    partial record AccountWithdrawRequest : IRequest<Account, 
}

file[1] = Account\Deposit\Account.generated.cs
// Auto-generated using OurPresence.Modeller template 'Business Logic Behaviour' version 1.0.0

// --------------------------------------------------------------------------------
// WARNING: This file will be overwritten if re-generation is triggered.
// --------------------------------------------------------------------------------

namespace Mizrael.SuperSafeBank.BusinessLogic.Account.Deposit
{
    partial record AccountDepositRequest : IRequest<Account, 
}

